### PR TITLE
Add test case for `internal/file`

### PR DESCRIPTION
**Pull request by hlts2**
Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

### Description:

I added test case for `internal/file`.

### Environment:

- Go Version: 1.14.3
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.11.5

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by kevindiu** [comment]
the test is fail..

```
% go test ./...
--- FAIL: TestOpen (0.00s)
    --- FAIL: TestOpen/returns_*os.File_when_path_is_`test/data` (0.00s)
        file_test.go:125: error = open test/data: permission denied
FAIL
FAIL	github.com/vdaas/vald/internal/file	2.045s
ok  	github.com/vdaas/vald/internal/file/watch	1.746s
FAIL
```

**Comment by kevindiu** [comment]
Please add comment to the `Open()` function

File: internal/file/file_test.go [hunk 1]
```diff
@@ -40,55 +39,82 @@ func TestOpen(t *testing.T) {
...
-		       args: args {
-		           path: "",
-		           flg: 0,
-		           perm: nil,
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
+		{
```
Comment (kevindiu) [inline internal/file/file_test.go:59]: I think you should also need to test if the directory is not exists for multiple hierarchy.

**Comment by vankichi** [comment]
@hlts2 Could you please add comment of `Open()`?
https://github.com/vdaas/vald/blob/master/internal/file/file.go#L25

File: internal/file/file.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ import (
 	"path/filepath"
 )
 
+// Open opens file with a gien path and flag and permission. If file does not exist, create the file.
```
Comment (kevindiu) [inline internal/file/file.go:25]:
```suggestion
// Open opens the file with the given path, flag and permission. 
// If the folder does not exists, create the folder.
// If the file does not exist, create the file. 
```

**Pull request merged**